src/flamenco/runtime/fd_runtime.c

```diff
@@ -1315,6 +1315,135 @@ fd_runtime_block_verify_ticks( fd_blockstore_t * blockstore,
+
+      if( FD_UNLIKELY( fd_txn_writes_query( acct_map, *readonly_acc, NULL ) ) ) {
+        err = FD_TXN_CONFLICT_READ_WRITE;
+        goto cleanup;
+      }
+    }
+
+    /* Iterate all read-only accounts in address lookup tables */
+    fd_txn_acct_addr_lut_t const * addr_luts = fd_txn_get_address_tables_const( TXN(txn) );
+    for( uchar j=0; j<TXN(txn)->addr_table_lookup_cnt; j++ ) {
+      fd_txn_acct_addr_lut_t const * addr_lut = &addr_luts[j];
+      fd_pubkey_t const *        addr_lut_acc = (fd_pubkey_t *)((uchar *)txn->payload + addr_lut->addr_off );
+      fd_funk_rec_key_t          funk_rec_key = fd_acc_funk_key( addr_lut_acc );
+      fd_funk_rec_t const *      addr_lut_rec = fd_funk_rec_query( funk, funk_txn, &funk_rec_key );
+      void *                              val = fd_funk_val( addr_lut_rec, fd_funk_wksp( funk ) );
+
+      fd_acct_addr_t * readonly_accts = (fd_acct_addr_t *) fd_type_pun((uchar *)val + addr_lut->readonly_off);
+      for( uchar k=0; k<addr_lut->writable_cnt; k++ ) {
+        if( FD_UNLIKELY( fd_txn_writes_query( acct_map, readonly_accts[k], NULL ) ) ) {
+          err = FD_TXN_CONFLICT_READ_WRITE;
+          goto cleanup;
+        }
+      }
+    }
+  }
+
+cleanup:
+  for( ulong i=0; i<txn_cnt; i++ ) {
+    fd_txn_p_t *              txn = txns + i;
+    fd_acct_addr_t * account_keys = (fd_acct_addr_t *)fd_type_pun( (void*)fd_txn_get_acct_addrs( TXN(txn), txn->payload ) );
+
+    for( fd_txn_acct_iter_t j=fd_txn_acct_iter_init( TXN(txn), FD_TXN_ACCT_CAT_WRITABLE );
+         j!=fd_txn_acct_iter_end();
+         j=fd_txn_acct_iter_next( j ) ) {
+      fd_acct_addr_t * writable_acc = &account_keys[ fd_txn_acct_iter_idx( j ) ];
+      fd_txn_writes_t * ele = fd_txn_writes_query( acct_map, *writable_acc, NULL );
+      if( FD_LIKELY( ele ) ) fd_txn_writes_remove( acct_map, ele );
+    }
+
+    /* Iterate all writable accounts in address lookup tables and remove from acct_map */
+    fd_txn_acct_addr_lut_t const * addr_luts = fd_txn_get_address_tables_const( TXN(txn) );
+    for( uchar j=0; j<TXN(txn)->addr_table_lookup_cnt; j++ ) {
+      fd_txn_acct_addr_lut_t const * addr_lut = &addr_luts[j];
+      fd_pubkey_t const *        addr_lut_acc = (fd_pubkey_t *)((uchar *)txn->payload + addr_lut->addr_off );
+      fd_funk_rec_key_t          funk_rec_key = fd_acc_funk_key( addr_lut_acc );
+      fd_funk_rec_t const *      addr_lut_rec = fd_funk_rec_query( funk, funk_txn, &funk_rec_key );
+      void *                              val = fd_funk_val( addr_lut_rec, fd_funk_wksp( funk ) );
+
+      fd_acct_addr_t * writable_accts = (fd_acct_addr_t *) fd_type_pun((uchar *)val + addr_lut->writable_off);
+      for( uchar k=0; k<addr_lut->writable_cnt; k++ ) {
+        fd_txn_writes_t * ele = fd_txn_writes_query( acct_map, writable_accts[k], NULL );
+        if( FD_LIKELY( ele ) ) fd_txn_writes_remove( acct_map, ele );
+      }
+    }
+  }
+  return err;
+}
```

Comment:
instead of doing it like this and re-parsing and re-querying everything you can just clear the map at the beginning of the function